rust/onnxruntime-sys/build.rs

```diff
@@ -0,0 +1,420 @@
+                );
+            }
+        }
+    }
+}
+
+fn prebuilt_archive_url() -> (PathBuf, String) {
+    let triplet = Triplet {
+        os: env::var("CARGO_CFG_TARGET_OS")
+            .expect("Unable to get TARGET_OS")
+            .parse()
+            .unwrap(),
+        arch: env::var("CARGO_CFG_TARGET_ARCH")
+            .expect("Unable to get TARGET_ARCH")
+            .parse()
+            .unwrap(),
+        accelerator: env::var(ORT_RUST_ENV_GPU)
+            .unwrap_or_default()
+            .parse()
+            .unwrap(),
+    };
+
+    let prebuilt_archive = format!(
+        "onnxruntime-{}-{}.{}",
+        triplet.as_onnx_str(),
+        ORT_VERSION,
+        triplet.os.archive_extension()
+    );
+    let prebuilt_url = format!(
+        "{}/v{}/{}",
+        ORT_RELEASE_BASE_URL, ORT_VERSION, prebuilt_archive
+    );
+
+    (PathBuf::from(prebuilt_archive), prebuilt_url)
+}
+
+fn prepare_libort_dir_prebuilt() -> PathBuf {
+    let (prebuilt_archive, prebuilt_url) = prebuilt_archive_url();
+
+    let out_dir = PathBuf::from(env::var("OUT_DIR").unwrap());
+    let extract_dir = out_dir.join(ORT_PREBUILT_EXTRACT_DIR);
+    let downloaded_file = out_dir.join(&prebuilt_archive);
+
+    println!("cargo:rerun-if-changed={}", downloaded_file.display());
+
+    if !downloaded_file.exists() {
+        println!("Creating directory {:?}", out_dir);
+        fs::create_dir_all(&out_dir).unwrap();
+
+        println!(
+            "Downloading {} into {}",
+            prebuilt_url,
+            downloaded_file.display()
+        );
+        download(&prebuilt_url, &downloaded_file);
+    }
+
+    if !extract_dir.exists() {
+        println!("Extracting to {}...", extract_dir.display());
+        extract_archive(&downloaded_file, &extract_dir);
+    }
+
+    extract_dir.join(prebuilt_archive.file_stem().unwrap())
+}
+
+fn prepare_libort_dir() -> PathBuf {
+    let strategy = env::var(ORT_RUST_ENV_STRATEGY);
+    println!(
+        "strategy: {:?}",
+        strategy.as_ref().map_or_else(|_| "unknown", String::as_str)
+    );
+    match strategy.as_ref().map(String::as_str) {
+        Ok("download") => prepare_libort_dir_prebuilt(),
+        Ok("system") => PathBuf::from(match env::var(ORT_RUST_ENV_SYSTEM_LIB_LOCATION) {
+            Ok(p) => p,
+            Err(e) => {
+                panic!(
+                    "Could not get value of environment variable {:?}: {:?}",
+                    ORT_RUST_ENV_SYSTEM_LIB_LOCATION, e
+                );
+            }
+        }),
+        Ok("compile") | Err(_) => prepare_libort_dir_compiled(),
+        _ => panic!("Unknown value for {:?}", ORT_RUST_ENV_STRATEGY),
+    }
+}
+
+fn prepare_libort_dir_compiled() -> PathBuf {
+    let mut config = cmake::Config::new("../../cmake");
+
+    config.define("onnxruntime_BUILD_SHARED_LIB", "ON");
+
+    if env::var(ORT_RUST_ENV_GPU).unwrap_or_default().parse() == Ok(Accelerator::Gpu) {
```

Comment:
1. I guess it is better to make [triplet](https://github.com/microsoft/onnxruntime/pull/12606/files#diff-a8903875d7a7d401f09518758fe7c764b68f3d8360f4ba30484384d69569a315R330-R343) to static variable and here expression refer to that static variable.
2. I guess it is better to base the accelerator decision on the [cargo feature](https://doc.rust-lang.org/cargo/reference/features.html#features) rather than via environment variable. like a `cuda` feature.

Reply:
I'm not sure I understand. I think using cargo features for build decisions is the wrong way to go. Cargo features should usually only refer to Rust code decisions.

Reply:
I think the build decisions affect some API code.
All API codes are same now. However,  make a difference code in the near future by started support directml build.
e.g. `OrtSessionOptionsAppendExecutionProvider_DML` function is only provided on directml build.
Then there was the need to control cargo feature as `directml` in [voicevox/onnxruntime-rs](https://github.com/VOICEVOX/onnxruntime-rs/blob/master/onnxruntime/src/session.rs#L339-L361) .
Therefore, I think better controlling build decisions  on cargo feature.